andi/andi.py

```diff
@@ -7,11 +9,40 @@
 from andi.typeutils import get_union_args, is_union
 
 
+def _get_globalns_as_get_type_hints(func) -> Dict:
+    """ Global namespace resolution extracted from ``get_type_hints`` method """
+    if isinstance(func, types.ModuleType):
+        return func.__dict__
+    else:
+        nsobj = func
+        # Find globalns for the unwrapped object.
+        while hasattr(nsobj, '__wrapped__'):
+            nsobj = nsobj.__wrapped__
+        return getattr(nsobj, '__globals__', {})
+
+
+def _get_globalns_for_attrs(func) -> Dict:
+    """ Adds partial support for postponed type annotations in attrs classes.
+    Also required to support attrs classes when
+    ``from __future__ import annotations`` is used (default for python 4.0).
+    See https://github.com/python-attrs/attrs/issues/593 """
+    return dict(sys.modules[func.__module__].__dict__)
+
+
+def _get_globalns(func) -> Dict:
+    """ Returns the global namespace that will be used for the resolution
+    of postponed type annotations """
+    ns = dict(_get_globalns_for_attrs(func))
+    ns.update(_get_globalns_as_get_type_hints(func))
+    return ns
+
+
 def inspect(func: Callable) -> Dict[str, List[Optional[Type]]]:
```

Comment:
@kmike an alternative to fix the attrs problem would be to provide a inspect for classes. Something like:

```py
def inspect_cls(cls: type):
    ...
   annotations = get_type_hints(cls.__init__, cls.__globals__)
    ...
```

In this case I think we would have access to the global namespace easily. What do you think?